atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -301,6 +301,10 @@ private enum State {
 
     protected final Closer closer = Closer.create();
 
+    private final SnapshotTransactionMetricFactory snapshotTransactionMetricFactory;
+
+    private final KeyValueSnapshotEventRecorder snapshotEventRecorder;
```

Comment:
I might be missing something, but why are we now using two objects with such different APIs when it comes to emitting metrics? I'd trend towards having a consistent use of something similar to the EventRecorder: an interface that is semantic oriented and abstracts away the inner details of composing the metric.

Of course here we can't totally get away with abstracting the impl away, since we do indeed case about the timers, but there is probably a more elegant way to do this? Probably move a lot of the `try timed` statements to an event recorder class and also give it some update methods?

Reply:
That's a fair concern: I extracted what was necessary for the read path without (significantly) modifying the other usages in SnapshotTransaction. This was more of a bandwidth thing, I can track a separate task for this on internal tracking tool?

Reply:
Yeah, happy to do it in a follow up PR, but thing we really should do it. It would be quite sad if that was the state we left things when having the chance and time to do a quite big refactor